sunsynk/sensor.py

```diff
@@ -241,17 +241,90 @@ def update_value(self) -> None:
             _LOGGER.error("Could not decode temperature: %s", err)
 
 
+class Minutes(int):
+    """Deals with inverter time format conversion complexities."""
+
+    @classmethod
+    def from_str_value(cls, str_value: str) -> Minutes:
+        """Create from a string in hh:mm format."""
+        return cls(int(str_value[:-3]) * 60 + int(str_value[-2:]))
```

Comment:
A suggestion for a property setter below

```suggestion
    @str_value.setter
    def str_value(self, value: str):
        """Parse a string in hh:mm format."""
        (hour, _, min) = value.partition(":") 
        self.minutes = int(hour) * 60 + int(min)
```

Inheriting from a base value like `int` is not ideal, since the value will get replace on each assignment/math operation. So best to store the value of `minutes` in the class, thne it becomes `self.minutes`

```python
class Minutes:
    minutes: int
```

Reply:
Please check this